src/main/java/reposense/parser/ArgsParser.java

```diff
@@ -214,4 +217,26 @@ private static void verifyDatesRangeIsCorrect(Optional<Date> sinceDate, Optional
             throw new ParseException(MESSAGE_SINCE_DATE_LATER_THAN_UNTIL_DATE);
         }
     }
+
+    /**
+     * Verifies that {@code sinceDate} is no later than today's date.
+     * @throws ParseException if {@code sinceDate} is later than today's date.
+     */
+    private static void verifySinceDateIsValid(Optional<Date> sinceDate) throws ParseException {
+        long todayDate = new Date().getTime();
+        if (sinceDate.isPresent() && sinceDate.get().getTime() > todayDate) {
+            throw new ParseException(String.format(MESSAGE_DATE_OUT_OF_BOUND, "Since date"));
+        }
+    }
+
+    /**
+     * Verifies that {@code untilDate} is no later than today's date.
+     * @throws ParseException if {@code untilDate} is later than today's date.
+     */
+    private static void verifyUntilDateIsValid(Optional<Date> untilDate) throws ParseException {
+        long todayDate = new Date().getTime();
+        if (untilDate.isPresent() && untilDate.get().getTime() > todayDate) {
+            throw new ParseException(String.format(MESSAGE_DATE_OUT_OF_BOUND, "Until date"));
+        }
```

Comment:
Why is `verifyUntilDateIsValid` the same as `verifySinceDateIsValid`?

Reply:
My bad here. My thought was to make two methods with the same functionality but different logging messages. I should have made a single common method here to verify if the dates are after today's date.

But I'll remove these two methods as we are taking into account where users will want to set the configs to capture the repos ahead of time.